datafusion/sqllogictest/test_files/copy.slt

```diff
@@ -64,6 +64,25 @@ select * from validate_parquet;
 1 Foo
 2 Bar
 
+query ??
+COPY 
+(values (struct ('foo', (struct ('foo', make_array(1,2,3)))), make_array(timestamp '2023-01-01 01:00:01',timestamp '2023-01-01 01:00:01')), 
```

Comment:
New test case covers:

1. Struct types
2. Array types
3. Multiple levels of nesting (struct of struct with array value)
4. Timestamp types 

I attempted to add array of struct as specifically flagged in #8851, but was unable to construct the literal value due to #8867.

Reply:
Thank you. I left a comment there

Would it be possible to change the constant values of the two rows so it is clear they are different

e.g. 
```suggestion
(values (struct ('foo', (struct ('foo', make_array(10,20,30)))), make_array(timestamp '2024-01-01 01:00:01',timestamp '2024-01-01 01:00:01')), 
```

Reply:
I pushed up a change to make the rows distinct and add an array of struct as #8867 is fixed after pulling main.